protocol-units/bridge/contracts/src/AtomicBridgeInitator.sol

```diff
@@ -46,12 +57,15 @@ contract AtomicBridgeInitiator is IAtomicBridgeInitiator, Initializable {
         }
         // If msg.value is greater than 0, convert ETH to WETH
         if (ethAmount > 0) weth.deposit{value: ethAmount}();
-        //Transfer WETH to this contract, revert if transfer fails
+        // Transfer WETH to this contract, revert if transfer fails
         if (wethAmount > 0) {
             if (!weth.transferFrom(originator, address(this), wethAmount)) revert WETHTransferFailed();
         }
 
-        nonce++; //increment the nonce
+        // Update the balance of the originator 
+        balances[originator] += totalAmount;
```

Comment:
Interesting approach to track balances per recipient. Why not just treat the whole thing as a pool?

Reply:
When you say treat it as a pool. What data type do you propose? For liquidity pools I've always seen mappings of addresses to amounts.

Reply:
With pooled I mean, just throw all the ETH in one big bucket. Saves complexity and gas etc.

Reply:
hmm. Ok, lets give a try.